app/src/main/java/in/testpress/testpress/ui/MainActivity.java

```diff
@@ -186,21 +196,83 @@ public boolean onLongClick(View view) {
         });
     }
 
+    private void initializeNavigationHeaderView() {
+        View headerView = navigationView.getHeaderView(0);
+        navigationHeader = headerView.findViewById(R.id.navigation_header_container);
+        username = headerView.findViewById(R.id.tv_profile);
+        profilePicture = headerView.findViewById(R.id.img_profile);
+    }
+
+    private void getProfileDetailsFromViewModel() {
+        if (isUserAuthenticated) {
+            profileDetailsViewModel.get(serviceProvider, this).observe(this, new Observer<ProfileDetails>() {
+                @Override
+                public void onChanged(ProfileDetails profileDetails) {
+                    MainActivity.this.profileDetails = profileDetails;
+                    setProfileDetailsOnDrawer();
+                }
+            });
+        }
+    }
+
+    private void setProfileDetailsOnDrawer() {
+        if (profileDetails == null) {
+            hideNavigationHeader();
+        } else {
+            navigationHeader.setVisibility(View.VISIBLE);
+            username.setText(profileDetails.getDisplayName());
+            getBitmapFromString();
+            setOnClickListener();
+        }
+    }
+
+    private void hideNavigationHeader() {
+        navigationHeader.setVisibility(View.GONE);
+    }
+
+    private void getBitmapFromString() {
+        ImageLoader.getInstance().loadImage(profileDetails.getPhoto(), new SimpleImageLoadingListener() {
+            @Override
+            public void onLoadingFailed(String imageUri, View view, FailReason failReason) {
+                hideNavigationHeader();
```

Comment:
Do not hide the navigation header if image fetch is failed.  Show placeholder instead.